### PR TITLE
add staging db infra

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1219

### Change description ###

add staging db infra

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -16,6 +16,20 @@ module "reform-notifications-db" {
   subscription       = var.subscription
 }
 
+module "reform-notifications-staging-db" {
+  source             = "git@github.com:hmcts/cnp-module-postgres?ref=master"
+  product            = "${var.product}-staging-${var.component}"
```
Comment (nitinprabhu) [inline infrastructure/main.tf:21]: may be `${var.component}-staging` is sufficient  ?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -16,6 +16,20 @@ module "reform-notifications-db" {
...
+  location           = var.location_db
+  env                = var.env
+  database_name      = "notifications-staging"
```
Comment (nitinprabhu) [inline infrastructure/main.tf:24]: db name can be same without staging suffix as the host itself would be sufficient

Comment (doncem) [reply infrastructure/main.tf:24]: can we reuse main db and just define `database_name` for staging?

Comment (mdayican) [reply infrastructure/main.tf:24]: > can we reuse main db and just define `database_name` for staging?

I think not possible without changing terraform db module.

Comment (mdayican) [reply infrastructure/main.tf:24]: > db name can be same without staging suffix as the host itself would be sufficient

it is sufficient, I think better to keep?

Comment (mdayican) [reply infrastructure/main.tf]: changed the name

**Pull request merged**